File: Faza6/OnlineGamesShop/src/main/java/com/rich/onlinegamesshop/model/views/StoreView.java
package com.rich.onlinegamesshop.model.views;
import jakarta.persistence.Entity;
import jakarta.persistence.Id;
import jakarta.persistence.Table;
import lombok.AllArgsConstructor;
import lombok.Data;
import lombok.NoArgsConstructor;
import org.hibernate.annotations.Immutable;
import org.hibernate.annotations.View;

@Data
@Entity
@Immutable
@Table(name = "storeview")
@NoArgsConstructor
@AllArgsConstructor
public class StoreView {

    @Id
    private Integer id_game;
    private String title;
    private String genre;
    private String publisher_name;
    private String website;
    private String platform_name;
    private String manufacturer;
}

File: Faza6/FrontEnd/Angular/src/app/model/Views/StoreView.ts
export class StoreView{
    id_game:number;
    title:string;
    genre:string;
    image:string;
    publisher_name:string;
    website:string;
    platform_name:string;
    manufacturer:string;
    constructor(
      id_game:number,
      title:string,
      genre:string,
      image:string,
      publisher_name:string,
      website:string,
      platform_name:string,
      manufacturer:string,
    ){
        this.id_game=id_game;
        this.title=title;
        this.genre=genre;
        this.image=image;
        this.publisher_name=publisher_name;
        this.website=website;
        this.platform_name=platform_name;
        this.manufacturer=manufacturer;
    }
}
